internal/types/document.go

```diff
@@ -316,6 +334,60 @@ func (d *Document) RemoveByPath(path Path) {
 	removeByPath(d, path)
 }
 
+// setByPath sets value by given path.
+// If some parts of the path are missing, they will be created.
+// The Document type will be used to create these parts.
+func (d *Document) setByPath(path Path, value any) error {
+	innerComp, err := d.GetByPath(path.TrimSuffix())
+	if err != nil {
+		next := d
+
+		var insertedPath Path
+		for _, pathElem := range path.TrimSuffix().Slice() {
+			insertedPath = insertedPath.Append(pathElem)
+
+			v, err := d.GetByPath(insertedPath)
+			if err != nil {
+				suffix := len(insertedPath.Slice()) - 1
+				if suffix < 0 {
+					panic("invalid path")
+				}
+
+				must.NoError(next.Set(insertedPath.Slice()[suffix], must.NotFail(NewDocument())))
+
+				next = must.NotFail(d.GetByPath(insertedPath)).(*Document)
+			}
+
+			if doc, ok := v.(*Document); ok {
+				next = doc
+			}
+		}
+		innerComp = must.NotFail(d.GetByPath(path.TrimSuffix()))
+	}
+
+	switch inner := innerComp.(type) {
+	case *Document:
+		err := inner.Set(path.Suffix(), value)
+		if err != nil {
+			return err
+		}
+	case *Array:
+		index, err := strconv.Atoi(path.Suffix())
+		if err != nil {
+			return err
```

Comment:
Is this error possible at all? Is it a valid case if `strconv.Atoi(path.Suffix())` returns an error? Or should it be impossible (so we can panic instead of returning an error)?

Reply:
Yeah, I guess if we got `innerComp` from the `GetByPath` it must be array there.

Reply:
If this is a possible case, let's cover it with a test to check the behavior?